rust/lance/src/io/writer/statistics.rs

```diff
@@ -0,0 +1,1082 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Statistics collection utilities
+
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use crate::datatypes::Field;
+use crate::error::Result;
+use arrow_array::{
+    builder::{
+        make_builder, ArrayBuilder, BinaryBuilder, BooleanBuilder, PrimitiveBuilder, StringBuilder,
+    },
+    types::{
+        Date32Type, Date64Type, Decimal128Type, DurationMillisecondType, Float32Type, Float64Type,
+        Int16Type, Int32Type, Int64Type, Int8Type, TimestampMicrosecondType, UInt16Type,
+        UInt32Type, UInt64Type, UInt8Type,
+    },
+    Array, ArrayRef, ArrowNumericType, ArrowPrimitiveType, BinaryArray, BooleanArray,
+    Decimal128Array, DictionaryArray, Int32Array, Int64Array, PrimitiveArray, RecordBatch,
+    StringArray, StructArray,
+};
+use std::str;
+
+use arrow_schema::DataType;
+use arrow_schema::{ArrowError, Field as ArrowField, Schema as ArrowSchema};
+use datafusion::common::ScalarValue;
+use num_traits::bounds::Bounded;
+
+/// Max number of bytes that are included in statistics for binary columns.
+const BINARY_PREFIX_LENGTH: usize = 16;
```

Comment:
```suggestion
/// Max number of bytes that are included in statistics for binary columns.
const BINARY_PREFIX_LENGTH: usize = 64;
```

I think this should be larger. At the very least, it should be large enough to hold a UUID hyphenated string, which has 36 characters.